library/src/main/java/io/ona/kujaku/activities/MapActivity.java

```diff
@@ -101,79 +102,95 @@ protected void onCreate(Bundle savedInstanceState) {
-                String mapBoxAccessToken = bundle.getString(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN);
-                String[] stylesArray = bundle.getStringArray(Constants.PARCELABLE_KEY_MAPBOX_STYLES);
-                currentStylePath = "";
+            if (bundle != null) {
+                return bundle;
+            }
+        }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND)) {
-                    topLeftBound = bundle.getParcelable(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND);
+        return null;
+    }
 
-                    if (bundle.containsKey(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND)) {
-                        bottomRightBound = bundle.getParcelable(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND);
-                    }
-                }
+    private void initializeMapActivityAfterPermissionsSupplied(Bundle savedInstanceState) {
+        Bundle bundle = getIntentExtras();
+        if (bundle != null) {
+            String[] stylesArray = bundle.getStringArray(Constants.PARCELABLE_KEY_MAPBOX_STYLES);
+            currentStylePath = "";
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
-                    maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND)) {
+                topLeftBound = bundle.getParcelable(Constants.PARCELABLE_KEY_TOP_LEFT_BOUND);
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
-                    minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
+                if (bundle.containsKey(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND)) {
+                    bottomRightBound = bundle.getParcelable(Constants.PARCELABLE_KEY_BOTTOM_RIGHT_BOUND);
                 }
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
-                    cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MAX_ZOOM)) {
+                maxZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MAX_ZOOM);
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_ZOOM)) {
-                    cameraZoom = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_ZOOM);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_MIN_ZOOM)) {
+                minZoom = bundle.getDouble(Constants.PARCELABLE_KEY_MIN_ZOOM);
+            }
 
-                if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TILT)) {
-                    cameraTilt = bundle.getDouble(Constants.PARCELABLE_KEY_CAMERA_TILT);
-                }
+            if (bundle.containsKey(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG)) {
+                cameraTargetLatLng = bundle.getParcelable(Constants.PARCELABLE_KEY_CAMERA_TARGET_LATLNG);
```

Comment:
What about this https://www.mapbox.com/mapbox-gl-js/style-spec/#root-center

Reply:
I will look at this

Reply:
We can add this as an issue ❗️ ❗️